pyscript.core/test/fetch.html

```diff
@@ -24,51 +24,47 @@
         );
       </script>
       <script type="mpy" async>
-          import js, json
-          from pyscript import document, fetch
+        import js, json
+        from pyscript import document, fetch
 
-          fetch_text = await fetch("config.json").text()
-          if (fetch_text != js.fetch_text):
-            raise Exception("fetch_text")
+        fetch_text = await (await fetch("config.json")).text()
+        if (fetch_text != js.fetch_text):
+          raise Exception("fetch_text")
 
-          fetch_json = await fetch("config.json").json()
-          if (json.dumps(fetch_json).replace(" ", "") != js.fetch_json):
-            raise Exception("fetch_json")
+        fetch_json = await (await fetch("config.json")).json()
+        if (json.dumps(fetch_json).replace(" ", "") != js.fetch_json):
+          raise Exception("fetch_json")
 
-          fetch_buffer = await fetch("config.json").bytearray()
-          if (len(fetch_buffer) != js.fetch_buffer):
-            raise Exception("fetch_buffer")
+        fetch_buffer = await (await fetch("config.json")).arrayBuffer()
+        if (len(fetch_buffer) != js.fetch_buffer):
+          raise Exception("fetch_buffer")
 
-          print(await fetch("config.json").bytearray())
-          print(await fetch("config.json").blob())
+        print(await (await fetch("config.json")).bytearray())
+        print(await (await fetch("config.json")).blob())
 
-          try:
-            await fetch("shenanigans.nope").text()
-          except:
-            document.documentElement.classList.add('mpy')
+        if (await fetch("shenanigans.nope")).ok == False:
+          document.documentElement.classList.add('mpy')
       </script>
       <script type="py" async>
         import js, json
         from pyscript import document, fetch
 
-        fetch_text = await fetch("config.json").text()
+        fetch_text = await (await fetch("config.json")).text()
```

Comment:
You probably know this, but the double awaits like this is a bit unusual, typically we would do something like this in python:

```python
response = await fetch()..
result = await response.text()
```

Given that this is just for testing I don't think its a blocker

Reply:
it's just for test and trust me, in JS that's not unusual at all, it's the reason we wanted to shortcut in the first place. Please also read the issue description I've just updated as we might have the best from all worlds.